theano/gof/cmodule.py

```diff
@@ -640,480 +653,427 @@ def refresh(self, age_thresh_use=None, delete_if_problem=False):
+                    _rmtree(*a, **kw)
 
         _logger.debug('Time needed to refresh cache: %s',
                       (time.time() - start_time))
 
         return too_old_to_use
 
-    def module_from_key(self, key, fn=None, keep_lock=False, key_data=None):
+    def _get_from_key(self, key, key_data=None):
         """
-        :param fn: A callable object that will return an iterable object when
-        called, such that the first element in this iterable object is the
-        source code of the module, and the last element is the module itself.
-        `fn` is called only if the key is not already in the cache, with
-        a single keyword argument `location` that is the path to the directory
-        where the module should be compiled.
-
-        :param key_data: If not None, it should be a KeyData object and the
-        key parameter should be None. In this case, we use the info from the
-        KeyData object to recover the module, rather than the key itself. Note
-        that this implies the module already exists (and may or may not have
-        already been loaded).
+        Returns a module if the passed-in key is found in the cache
+        and None otherwise.
+
+        May raise ValueError if the key is malformed.
         """
-        # We should only use one of the two ways to get a module.
-        assert key_data is None or key is None
-        rval = None
+        name = None
         if key is not None:
+            assert key_data is None
             try:
                 _version, _rest = key
             except (TypeError, ValueError):
                 raise ValueError(
-                        "Invalid key. key must have form (version, rest)", key)
-        name = None
-        if key is not None and key in self.entry_from_key:
-            # We have seen this key either in this process or previously.
-            name = self.entry_from_key[key]
-        elif key_data is not None:
+                    "Invalid key. key must have form (version, rest)", key)
+            if key in self.entry_from_key:
+                name = self.entry_from_key[key]
+        else:
+            assert key_data is not None
             name = key_data.get_entry()
-        if name is not None:
-            # This is an existing module we can recover.
-            if name not in self.module_from_name:
-                _logger.debug('loading name %s', name)
-                self.module_from_name[name] = dlimport(name)
-                self.stats[1] += 1
-            else:
-                self.stats[0] += 1
-            _logger.debug('returning compiled module from cache %s', name)
-            rval = self.module_from_name[name]
+        if name is None:
+            return None
+        return self._get_module(name)
+
+    def _get_from_hash(self, module_hash, key, keep_lock=False):
+        if module_hash in self.module_hash_to_key_data:
+            _logger.debug("Duplicated module! Will re-use the "
```

Comment:
I think you should remove this logged entry. This method get called just to check if the module_hash in known to prevent recompilation. Why this should be a problem.